astropy/coordinates/spectral_coordinate.py

```diff
@@ -738,7 +602,7 @@ def with_observer_stationary_relative_to(self, frame, velocity=None, preserve_ob
                 # otherwise frame is ready to go
             else:
                 if velocity is None:
-                    differentials = CartesianDifferential(0 * u.m / u.s, 0 * u.m / u.s, 0 * u.m / u.s)
+                    differentials = ZERO_VELOCITIES
                 else:
                     differentials = CartesianDifferential(*velocity)
```

Comment:
This makes me wonder, why not allow the velocity to be specified as a `CartesianDifferential` in the first place? With a recent PR of mine, that will now be passed along, but you have to *not* expand the velocity (which is fine, a triplet can be used as input anyway). Anyway, bottom line, would suggest to remove the `*` here and update the docstring that a `CartesianDifferential` is allowed...

Reply:
And then below too work in terms of a differential... (since this is really rather off-topic, do feel free to ignore).

Reply:
I've adjusted it to allow ``CartesianDifferential`` to be passed in